### PR TITLE
fix: [CDS-87172]: fixed Corrupted YAMLs being stored on creation flow…

### PR DESCRIPTION
**Pull request by karansaraswat19**
… of service / environment / input sets

#### RCA
So, now we are maintaining a new field provider in the formik. We added the GitProviderSelect component in the GitsyncForm only so that we would see the cards for every entity which has gitx. Therefore, yaml also took up the provider field and corrupted it .

#### Solution
removing the git provider selector cards from everwhere other than pipeline, inputsets and clone pipeline flow. Move resources and edit git details flow also impacts other entities, so removed from there as well

#### Screenshots
Screenshot 2023-12-20 at 12 58 31 PM (2)
Screenshot 2023-12-20 at 12 58 39 PM (2)

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by karansaraswat19** [comment]
trigger messagecheck

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 1]
```diff
@@ -319,7 +323,7 @@ export function GitSyncForm<T extends GitSyncFormFields = GitSyncFormFields>(
   }
 
   useEffect(() => {
-    if (!formikProps.values.provider) {
+    if (!formikProps.values.provider && renderRepositoryLocationCard) {
```
Comment (rameshx) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:326]: Add `renderRepositoryLocationCard` as a dependency to this effect

File: src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts [hunk 2]
```diff
@@ -106,6 +106,7 @@ export function useSaveInputSet(inputSetInfo: InputSetInfo): UseSaveInputSetRetu
       conflictCommitId?: string
     }): CreateUpdateInputSetsReturnType => {
       let response: ResponseInputSetResponse | undefined = undefined
+      const inputSetYaml = yamlStringify({ inputSet: clearNullUndefined(omit(inputSetObj, 'provider')) })
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts:109]: @karansaraswat19 Have you tested InputSet edit flow after these changes?

Comment (karansaraswat19) [reply src/modules/70-pipeline/components/InputSetForm/useSaveInputSet.ts:109]: yes, sir tested

**Review by Eric761 — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**